Data access sealing

jongbonga opened the pull request.

jongbonga committed:
edit encryptBlob & encryptAndUploadFile functions

jongbonga committed:
seal data

jongbonga committed:
form validation

longtomjr reviewed (approved):
Looks good! Thanks!

PiDelport commented on src/components/AccessForm.vue:
```diff
@@ -1,55 +1,95 @@
+            JSON.stringify(this.form)
+          );
+          const enclavePublicKey = this.$store.getters.enclavePublicKey;
+          encryptBlob(encodedForm, enclavePublicKey);
```
Missing await here: we should probably at least log the result for now, and await to avoid silencing any unexpected error?

PiDelport commented on src/utils/cryptography.ts:
```diff
@@ -27,11 +27,10 @@ export interface EncryptedMessage {
  */
 export async function encryptBlob(
-  plainBlob: Blob,
+  plaintext: Uint8Array,
```
The name and docstring still reflects `Blob`: if we need both `Blob` and `Uint8Array` interfaces, it's probably better to split out a small wrapper function? (And probably one for `string`, too, to keep the trickier encoding logic in one place, rather than relying on view code to do it?)

PiDelport replied:
Cool, this should be addressed in 21f01e4.

PiDelport reviewed (changes requested):
Minor feedback on the crypto interface: @jongbonga, I can quickly take take of it, if you'd like?

jongbonga committed:
await encryptBlob when sealing

PiDelport committed:
refactor(cryptography): expose and use encryptBytes, encryptBlob, encryptJson

This lets the vie code call the higher-level interfaces, and keeps the
details and implementation choices around byte-level encoding in the
cryptography module.

PiDelport committed:
docs(cryptography): update and expand comment a bit

PiDelport committed:
docs(AccessForm): add note about unproxied form data

PiDelport reviewed (approved):
Cool, looks good! 🎉

Can you look over the cryptography module changes I made before merging?

longtomjr reviewed (approved):
Looks good! Lets get this merged in!
